File: packages/tests/importer/fixtures/walk/source.kt
data class Telemetry(
    val uptimeMs: Int,
    var status: String
)

class Config {
    val id: Long? = null
    var isActive: Boolean = false
}

class Matrix(
    val tags: List<String>,
    val bytes: ByteArray,
    val counts: IntArray
)

enum class Status { Idle, Active }

private class Hidden {}
class Visible {
    private val secret: Int = 1
    val publicValue: Int = 2
}

File: packages/tests/importer/fixtures/walk/source.ts
interface Telemetry {
	uptime_ms: number;
	status: string;
	tags: number[];
}

interface I {
	x: number;
	note?: string;
}

type Mode = "Idle" | "Active" | "Error" | "Offline";

enum Severity {
	Info,
	Warn,
	Error,
}

type Reading = { sensor_id: number; value: number };

/** Telemetry packet. */
interface TelemetryDoc {
	/** Monotonic clock. */
	uptime_ms: number;
	status: string;
}

type Either = Foo | Bar;

interface ArrayTest {
	xs: Array<number>;
}

export interface Exported {
	x: number;
}
